src/m365/aad/commands/app/app-list.spec.ts

```diff
@@ -0,0 +1,145 @@
+const command: Command = require('./app-list');
+
+describe(commands.APP_LIST, () => {
+  let log: string[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => {});
+    sinon.stub(pid, 'getProcessName').callsFake(() => undefined);
+    auth.service.connected = true;
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([request.get]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.APP_LIST), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it(`should get a list of Azure AD app registrations`, async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/applications`) {
+        return {
+          value: [
+            {
+              id: '340a4aa3-1af6-43ac-87d8-189819003952',
+              appId: '9b1b1e42-794b-4c71-93ac-5ed92488b67f',
+              displayName: 'My App 1',
+              description: 'My second app',
+              signInAudience: 'My Audience'
+            },
+            {
+              id: '340a4aa3-1af6-43ac-87d8-189819003953',
+              appId: '9b1b1e42-794b-4c71-93ac-5ed92488b670',
+              displayName: 'My App 2',
+              description: 'My second app',
+              signInAudience: 'My Audience'
+            }
+          ]
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {}
+    });
+
+    assert(
+      loggerLogSpy.calledWith([
+        {
+          id: '340a4aa3-1af6-43ac-87d8-189819003952',
+          appId: '9b1b1e42-794b-4c71-93ac-5ed92488b67f',
+          displayName: 'My App 1',
+          description: 'My second app',
+          signInAudience: 'My Audience'
+        },
+        {
+          id: '340a4aa3-1af6-43ac-87d8-189819003953',
+          appId: '9b1b1e42-794b-4c71-93ac-5ed92488b670',
+          displayName: 'My App 2',
+          description: 'My second app',
+          signInAudience: 'My Audience'
+        }
+      ])
+    );
+  });
+
+  it('handles error when retrieving app list failed', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/applications`) {
+        throw 'An error has occurred';
```

Comment:
It's unlikely that a plain error like this is thrown, let's throw a more realistic one.